backend/app/api/endpoints/admin/device_monitor.py

```
@@ -0,0 +1,282 @@
+    )
+
+
+@router.get("/devices", response_model=AdminDeviceListResponse)
+async def get_all_devices(
+    page: int = Query(1, ge=1, description="Page number"),
+    limit: int = Query(50, ge=1, le=100, description="Items per page"),
+    status: Optional[str] = Query(None, description="Filter by status"),
+    device_type: Optional[str] = Query(None, description="Filter by device type"),
+    bind_shell: Optional[str] = Query(None, description="Filter by bind shell"),
+    search: Optional[str] = Query(None, description="Search by device name or ID"),
+    db: Session = Depends(get_db),
```

Ruff (0.15.6)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)